grakn-dist/src/main/java/ai/grakn/dist/ProcessHandler.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Grakn - A Distributed Semantic Database
+ * Copyright (C) 2016  Grakn Labs Limited
+ *
+ * Grakn is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ */
+
+package ai.grakn.dist;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FilenameFilter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.stream.Stream;
+
+import static java.util.stream.Collectors.joining;
+
+/**
+ *
+ * @author Michele Orsi
+ */
+public class ProcessHandler {
+
+    OutputCommand executeAndWait(String[] cmdarray, String[] envp, File dir) {
+
+        StringBuffer outputS = new StringBuffer();
+        int exitValue = 1;
+
+        Process p;
+        BufferedReader reader = null;
+        try {
+            p = Runtime.getRuntime().exec(cmdarray, envp, dir);
+            p.waitFor();
+            exitValue = p.exitValue();
+            reader =
+                    new BufferedReader(new InputStreamReader(p.getInputStream(), StandardCharsets.UTF_8));
+
+            String line = "";
+            while ((line = reader.readLine()) != null) {
+                outputS.append(line + "\n");
+            }
+
+        } catch (InterruptedException | IOException e) {
+            e.printStackTrace();
+        } finally {
+            if (reader != null) {
+                try {
+                    reader.close();
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+        OutputCommand outputCommand = new OutputCommand(outputS.toString().trim(), exitValue);
+        return outputCommand;
+    }
+
+    String getPidFromFile(String fileName) {
+        String pid="";
+        if (Files.exists(Paths.get(fileName))) {
+            try {
+                pid = new String(Files.readAllBytes(Paths.get(fileName)),StandardCharsets.UTF_8).trim();
+            } catch (IOException e) {
+                // DO NOTHING
```

Comment:
Return `@Nullable String` or an `Optional<String>` for this. Hiding errors scares me :(